3-enrich/scala-common-enrich/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -264,12 +276,27 @@ object EnrichmentManager {
       }
     }
 
-    // Finally anonymize the IP address
+    // For anonymize the IP address
     Option(event.user_ipaddress).map(ip => event.user_ipaddress = registry.getAnonIpEnrichment match {
       case Some(anon) => anon.anonymizeIp(ip)
       case None => ip
     })
 
+    // Finalize the currency conversion
+    val currencyConversionFinal = {
+      registry.getCurrencyConversionEnrichment match {
+        case Some(currency) => {
+          val trAmounts  = (trTotal |@| trTax |@| trShipping ) {  
+                             TransactionAmounts(_, _, _)
+                           }
+          val currencyConversion = (trCurrency |@| trAmounts |@| tiCurrency |@| tiPrice) { 
+                                     currency.convertCurrencies(_, _, _, _)
+                                   }
+          currencyConversion                                
```

Comment:
Implicit return does not seems to be a member of Unit, thus causes problem with toValidaitonNel